JotunnLib/Extensions/GameObjectExtension.cs

```diff
@@ -104,6 +106,174 @@ public static GameObject OrNull(this GameObject @this)
+        /// <param name="gameObject"></param>
+        /// <param name="componentNames"></param>
+        /// <returns></returns>
+        public static bool HasAllComponents(this GameObject gameObject, params string[] componentNames)
+        {
+            foreach (var name in componentNames)
+            {
+                if (gameObject.GetComponent(name) == null)
+                {
+                    return false;
+                }
+            }
+            return true;
+        }
+
+        /// <summary>
+        ///     Check if GameObject has all of the specified components.
+        /// </summary>
+        /// <param name="gameObject"></param>
+        /// <param name="components"></param>
+        /// <returns></returns>
+        public static bool HasAllComponents(this GameObject gameObject, params Type[] components)
+        {
+            foreach (var compo in components)
+            {
+                if (gameObject.GetComponent(compo) == null)
+                {
+                    return false;
+                }
+            }
+            return true;
+        }
+
+        /// <summary>
+        ///     Check if GameObject or any of it's children
+        ///     have any of the specified components.
+        /// </summary>
+        /// <param name="gameObject"></param>
+        /// <param name="components"></param>
+        /// <returns></returns>
+        public static bool HasAnyComponentInChildren(
+            this GameObject gameObject,
+            bool includeInactive = false,
+            params Type[] components
+        )
+        {
+            foreach (var compo in components)
+            {
+                if (gameObject.GetComponentInChildren(compo, includeInactive) != null)
+                {
+                    return true;
+                }
+            }
+            return false;
+        }
+
+        /// <summary>
+        ///     Check if GameObject or any of it's children
+        ///     have the specific component.
+        /// </summary>
+        /// <param name="gameObject"></param>
+        /// <param name="includeInactive">Whether to include inactive child objects in the search or not.</param>
+        /// <returns></returns>
+        public static bool HasComponentInChildren<T>(this GameObject gameObject, bool includeInactive = false) where T : Component
+        {
+            return gameObject.GetComponentInChildren<T>(includeInactive) != null;
+        }
+
+        /// <summary>
+        ///     Extension method to get the first component in the GameObject
+        ///     or it's children that has the specified name.
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <param name="gameObject"></param>
+        /// <param name="name"></param>
+        /// <param name="includeInactive">Whether to include inactive child objects in the search or not.</param>
+        /// <returns></returns>
+        public static T GetComponentInChildrenByName<T>(
```

Comment:
I'm not sure what for the use case for this method is, why is the component serched by T and checked with a name?

Reply:
It might honestly be too niche, I made it because there are a few cases in MoreVanillaBuildPrefabs where I need to search for a Component in a specific child object but I may not know what the transform tree looks like. I think I wrote it before I figured out how to search for a specific nested child or that Utils.FindChild existed. I'll update the PR to remove this one since it's better accomplished by using Utils.FindChild(name).GetComponent<T>() instead.